File: 2_7/2_7.go

package main

import (
	"fmt"
)

func findDestination(destinations [][]string) string {
	m := make(map[string]string)
	for _, p := range destinations {
		if p[0] == p[1] {
			return "обратитесь к специалисту"
		}
		m[p[0]] = p[1]
	}

	current := destinations[0][0]
	for {
		next, ok := m[current]
		if !ok {
			break
		}
		current = next
	}
	return current
}

func main() {
	destinations := [][]string{{"Новосибирск", "Дубай"}, {"Новосибирск", "Искитим"}, {"Искитим", "Линево"}}
	fmt.Println("Output:", findDestination(destinations))
}

File: 2_7/2_7.py

from typing import List, Tuple

def find_destination(destinations: List[Tuple[str, str]]) -> str:
    destination_map = {}
    for source, dest in destinations:
        if source == dest:
            return "обратитесь к специалисту"
        destination_map[source] = dest

    current = destinations[0][0]
    while current in destination_map:
        current = destination_map[current]

    return current


destinations = [("Новосибирск", "Дубай"), ("Новосибирск", "Искитим"), ("Искитим", "Линево")]
print("Output:", find_destination(destinations))

